### PR TITLE
return error on multibase and version

### PR DESCRIPTION
**Pull request by susarlanikhilesh**

### CODE REVIEW

**Review by Jorropo — APPROVED** [review]

**Review by Jorropo — CHANGES_REQUESTED** [review]
Could you please a unit test <3 ?

File: core/commands/cid.go [hunk 1]
```diff
@@ -85,6 +85,11 @@ The optional format string is a printf style format string:
 			if opts.newCodec != 0 && opts.newCodec != cid.DagProtobuf {
 				return fmt.Errorf("cannot convert to CIDv0 with any codec other than dag-pb")
 			}
+
+			if baseStr != "" {
+				return fmt.Errorf("cannot convert to CIDv0 with any multibase other than the implicit base58btc")
```
Comment (lidel) [inline core/commands/cid.go:90]: If i pass `base58btc` it will error, while it is a valid base for CIDv0 – please fix + add test :pray:

**Review by lidel — CHANGES_REQUESTED** [review]

File: core/commands/cid.go (hunk 1 above)
Comment (hacdias) [reply core/commands/cid.go:90]: Ping @susarlanikhilesh

**Comment by lidel** [comment]
Triage notes:
- Closing duye to bug describe in https://github.com/ipfs/kubo/pull/9968#discussion_r1272280809
- @susarlanikhilesh  Feel free to reopen as new PR  once fixed and has tests :pray:

**Pull request closed without merging**